src/main/java/xreliquary/items/ItemLanternOfParanoia.java

```diff
@@ -68,16 +71,25 @@ public void onUpdate(ItemStack stack, World world, Entity e, int i, boolean f) {
 			int playerY = MathHelper.floor(player.getEntityBoundingBox().minY);
 			int playerZ = MathHelper.floor(player.posZ);
 
+			BlockPos pos;
+			IBlockState state;
+			Block block;
+
 			placement:
 			for (int xDiff = -getRange(); xDiff <= getRange(); xDiff++) {
 				for (int zDiff = -getRange(); zDiff <= getRange(); zDiff++) {
 					for (int yDiff = getRange() / 2; yDiff >= -getRange() / 2; yDiff--) {
 						int x = playerX + xDiff;
 						int y = playerY + yDiff;
 						int z = playerZ + zDiff;
-						if (!player.world.isAirBlock(new BlockPos(x, y, z)))
+
+						pos = new BlockPos(x, y, z);
+						state = world.getBlockState(pos);
+						block = state.getBlock();
+
+						if (block instanceof BlockLiquid || block instanceof IFluidBlock || !block.isAir(state, world, pos) && !block.isReplaceable(world, pos))
 							continue;
-						int lightLevel = player.world.getLightFromNeighbors(new BlockPos(x, y, z));
+						int lightLevel = player.world.getLightFromNeighborsFor(EnumSkyBlock.BLOCK, pos);
```

Comment:
getLightFromNeighborsFor is actually client side only so this would crash on dedicated server. Would need to be replaced with something that works there.

Reply:
Thanks for pointing this out! It's completely my bad for failing to test in a dedicated environment (which I now have worked out how to do properly and have set up, so my most recent commit was tested with integrated and dedicated servers). 

From what I can tell, world.getLightFor is almost functionally identical to this check, except that it simply checks the individual block position rather than getting an average across its neighbours. I'll add additional comments at the end of this.